src/aap_eda/api/serializers/credential_type.py

```diff
@@ -53,10 +53,36 @@ class CredentialTypeCreateSerializer(serializers.ModelSerializer):
     )
 
     def validate(self, data):
-        if data.get("injectors") and data.get("inputs"):
-            errors = validate_injectors(
-                data.get("inputs"), data.get("injectors")
-            )
+        injectors = data.get("injectors")
+        inputs = data.get("inputs")
+
+        if self.partial:
+            # updating allow injectors/inputs be None, but not empty dict
+            if injectors is not None and not bool(injectors):
+                raise serializers.ValidationError(
+                    "Injectors field cannot be empty"
+                )
+
+            if inputs is not None and not bool(inputs):
+                raise serializers.ValidationError(
+                    "inputs field cannot be empty"
```

Comment:
Capitalize `Inputs` to match other error messages, or change to lower case for all messages.